packages/gatsby/src/recipes/providers/npm/package.js

```diff
@@ -0,0 +1,65 @@
+const execa = require(`execa`)
+const _ = require(`lodash`)
+
+const getPackageNames = packages => packages.map(n => n.name)
+const asyncForEach = async (array, callback) => {
+  for (let index = 0; index < array.length; index++) {
+    await callback(array[index], index, array)
+  }
+}
+
+const create = async ({ root }, packages) => {
+  const types = _.groupBy(packages, c => c.dependencyType)
+
+  // Run install for each dependency type
+  await asyncForEach(Object.keys(types), async type => {
+    let typeFlag = ``
+    if (type === `dev`) {
+      typeFlag = `--dev`
+    }
+    const command = [
+      `yarn`,
+      [`add`, `-W`, typeFlag, ...getPackageNames(types[type])],
```

Comment:
Why `-W` everywhere? Sounds dangerous.

Reply:
It's a yarnism that makes installs work properly in workspaces. It's a noopt in regular repos. We'll be moving this to look at what the user has set as their preferred client (npm vs. yarn) for release.